From 2e0c3105c62bfe01b54f6830030a231d15c378ba Mon Sep 17 00:00:00 2001
From: "Srivatsa S. Bhat (VMware)" <srivatsa@csail.mit.edu>
Date: Tue, 5 Jul 2022 10:00:00 -0700
Subject: [PATCH] Do not install auxiliary module from ice driver sources

Both iavf-4.5.3 and ice-1.9.11 driver sources include a backported
kernel module called intel_auxiliary.ko, which they both depend on. Since we
just need one copy of intel_auxiliary.ko in the final rpm, install it only
from the iavf driver sources and skip the installation from the ice
driver.

Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
 src/common.mk | 46 +++++++++++++++++++++++-----------------------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/src/common.mk b/src/common.mk
index df5e4fd..93d9f80 100644
--- a/src/common.mk
+++ b/src/common.mk
@@ -427,32 +427,32 @@ endif # check_aux_bus exists

 # The out-of-tree auxiliary module we ship should be moved into this
 # directory as part of installation.
-export INSTALL_AUX_DIR ?= updates/drivers/net/ethernet/intel/auxiliary
+#export INSTALL_AUX_DIR ?= updates/drivers/net/ethernet/intel/auxiliary

 # If we're installing auxiliary bus out-of-tree, the following steps are
 # necessary to ensure the relevant files get put in place.
-AUX_BUS_HEADER ?= linux/auxiliary_bus.h
-ifeq (${NEED_AUX_BUS},2)
-define auxiliary_post_install
-	install -D -m 644 Module.symvers ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/intel_auxiliary.symvers
-	install -d ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}
-	mv -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_MOD_DIR}/intel_auxiliary.ko* \
-	      ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/
-	install -D -m 644 ${AUX_BUS_HEADER} ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
-endef
-else
-auxiliary_post_install =
-endif
-
-ifeq (${NEED_AUX_BUS},2)
-define auxiliary_post_uninstall
-	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/intel_auxiliary.symvers
-	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/intel_auxiliary.ko
-	rm -f ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
-endef
-else
-auxiliary_post_uninstall =
-endif
+#AUX_BUS_HEADER ?= linux/auxiliary_bus.h
+#ifeq (${NEED_AUX_BUS},2)
+#define auxiliary_post_install
+#	install -D -m 644 Module.symvers ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/intel_auxiliary.symvers
+#	install -d ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}
+#	mv -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_MOD_DIR}/intel_auxiliary.ko* \
+#	      ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/
+#	install -D -m 644 ${AUX_BUS_HEADER} ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
+#endef
+#else
+#auxiliary_post_install =
+#endif
+
+#ifeq (${NEED_AUX_BUS},2)
+#define auxiliary_post_uninstall
+#	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/extern-symvers/intel_auxiliary.symvers
+#	rm -f ${INSTALL_MOD_PATH}/lib/modules/${KVER}/${INSTALL_AUX_DIR}/intel_auxiliary.ko
+#	rm -f ${INSTALL_MOD_PATH}/${KSRC}/include/linux/auxiliary_bus.h
+#endef
+#else
+#auxiliary_post_uninstall =
+#endif

 ifeq (${NEED_AUX_BUS},2)
 EXTRA_CFLAGS += -DUSE_INTEL_AUX_BUS
--
2.35.6

